packages/shared/src/utils/helpers/duration.ts

```diff
@@ -37,44 +76,48 @@ export const buildDurationConfig = (contract, durations = { min_max: {}, units_d
             if (
                 u !== 'd' && // when the expiray_type is intraday, the supported units are seconds, minutes and hours.
                 arr_units.indexOf(u) === -1 &&
-                duration_maps[u].order >= duration_maps[obj_min.unit].order &&
-                duration_maps[u].order <= duration_maps[obj_max.unit].order
+                duration_maps[u as TDurationMaps].order >= duration_maps[obj_min.unit as TDurationMaps].order &&
+                duration_maps[u as TDurationMaps].order <= duration_maps[obj_max.unit as TDurationMaps].order
             ) {
                 arr_units.push(u);
             }
         });
     }
 
-    durations.units_display[contract.start_type] = arr_units
-        .sort((a, b) => (duration_maps[a].order > duration_maps[b].order ? 1 : -1))
-        .reduce((o, c) => [...o, { text: duration_maps[c].display, value: c }], []);
+    duration_units = arr_units
+        .sort((a, b) => (duration_maps[a as TDurationMaps].order > duration_maps[b as TDurationMaps].order ? 1 : -1))
+        .reduce((o, c) => [...o, { text: duration_maps[c as TDurationMaps].display, value: c }], [] as TUnit[]);
 
     return durations;
 };
 
-export const convertDurationUnit = (value, from_unit, to_unit) => {
-    if (!value || !from_unit || !to_unit || isNaN(parseInt(value))) {
+export const convertDurationUnit = (value: number, from_unit: string, to_unit: string) => {
+    if (!value || !from_unit || !to_unit || isNaN(value)) {
         return null;
     }
 
     const duration_maps = getDurationMaps();
 
-    if (from_unit === to_unit || !('to_second' in duration_maps[from_unit])) {
+    if (from_unit === to_unit || !('to_second' in duration_maps[from_unit as keyof typeof duration_maps])) {
         return value;
     }
 
-    return (value * duration_maps[from_unit].to_second) / duration_maps[to_unit].to_second;
+    return (
+        (value * duration_maps[from_unit as keyof typeof duration_maps].to_second) /
+        duration_maps[to_unit as keyof typeof duration_maps].to_second
+    );
 };
 
-const getDurationFromString = duration_string => {
-    const duration = duration_string.toString().match(/[a-zA-Z]+|[0-9]+/g);
+const getDurationFromString = (duration_string: string) => {
+    const duration = duration_string.toString().match(/[a-zA-Z]+|[0-9]+/g) || '';
     return {
         duration: +duration[0], // converts string to numbers
         unit: duration[1],
     };
 };
 
-export const getExpiryType = store => {
+// TODO will change this after the global stores types get ready
+export const getExpiryType = (store: any) => {
```

Comment:
<img width="928" alt="Screenshot 2022-10-10 at 10 30 01" src="https://user-images.githubusercontent.com/103177211/194816920-b65db367-bc74-4b28-b9cb-70d8d014c7b3.png">
@niloo-fs please address this too 🙂
```
((toMoment(expiry_date) as unknown as moment.Moment).isAfter(
            toMoment(server_time) as unknown as moment.MomentInput,
            'day'
        ) ||
            !hasIntradayDurationUnit(duration_units_list));
```

Reply:
I changed this!
but what is this for?

Reply:
just to avoid TS warning about .isAfter() 🙂 Thank you 👍🏻